runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/quarkus/CassandraClientProducer.java

```diff
@@ -68,8 +73,29 @@ public QuarkusCqlSession createCassandraClient(QuarkusCqlSessionState quarkusCql
             .withMetricRegistry(metricRegistry)
             .withQuarkusEventLoop(mainEventLoop)
             .withConfigLoader(configLoaderBuilder.build());
-    quarkusCqlSessionState.setInitialized();
-    return builder.build();
+    return builder
+        .buildAsync()
+        .whenComplete(
+            (res, ex) -> {
+              if (ex == null) {
+                LOG.debug("Setting the QuarkusCqlSessionState to initialized.");
+                quarkusCqlSessionState.setInitialized();
+              }
+            });
+  }
+
+  @Produces
+  @ApplicationScoped
+  @Unremovable
+  public QuarkusCqlSession createCassandraClient(CompletionStage<QuarkusCqlSession> sessionFuture) {
+    LOG.trace("Produce QuarkusCqlSession bean.");
+    if (!config.cassandraClientInitConfig.eagerSessionInit) {
+      LOG.info(
+          "Injecting QuarkusCqlSession and setting eager-session-init = false may cause deadlock on the Vert.x thread. "
+              + "Please set it to true, inject only CompletionStage<QuarkusCqlSession>, "
+              + "or assert that initializing of the QuarkusCqlSession in your Application is happening on your thread pool (not Vert.x event loop).");
+    }
```

Comment:
To the point by @adutra above... this is only true for Vert.x reactive routes (since those are the only ones that run on the Vert.x thread pool).  Shouldn't we at least mention that constraint in the message?  Ideally we'd actually test for the usage of reactive routes in the app and only warn if they're actually in use... but failing that mentioning them explicitly seems like a good idea.

Reply:
I think that this part: `or assert that initializing of the QuarkusCqlSession in your Application is happening not on the Vert.x event loop.` consists of the case when the user is not using Vert.x - the end-user can assert that initialization is happening not on Vert.x event loop

Reply:
Yeah, _I_ understand that those things are connected... but it's not immediately clear to me that a general Quarkus user will connect "something happening on a vert.x event loop" to "I'm using reactive routes".  Regardless it's not a big concern.